AptosSDK/Src/TokenClient.cs

```diff
@@ -0,0 +1,367 @@
+			{
+				Type = TransactionPayloadTypes.EntryFunction,
+				Function = FunctionTypes.ClaimScript,
+				TypeArguments = Array.Empty<string>(),
+				Arguments = new object[]
+					{ sender, creator, collectionName, name, propertyVersion.ToString() }
+			};
+
+			var receipt = await GenerateSignSubmitTransaction(account, payload, extraArgs);
+
+			return receipt;
+		}
+
+		public async Task<CollectionPayload> GetCollectionData(string creator, string collectionName)
+		{
+			var collections = await _client.GetAccountResource(creator, ResourcesTypes.Collections);
+			var collectionData = collections.Data.ToObject<CollectionsResource>();
+			var handle = collectionData.CollectionData.Handle;
+
+			var request = new TableItemRequest
+			{
+				KeyType = KeyTypes.String,
+				ValueType = ValueTypes.CollectionData,
+				Key = collectionName
+			};
+
+			return await _client.GetTableItem<CollectionPayload>(handle, request);
+		}
+		
+		public async Task<TokenPayload> GetTokenData(string creator, string collectionName, string tokenName)
+		{
+			var collections = await _client.GetAccountResource(creator, ResourcesTypes.Collections);
+			var collectionData = collections.Data.ToObject<CollectionsResource>();
+			var handle = collectionData.TokenData.Handle;
+
+			var request = new TableItemRequest
+			{
+				KeyType = KeyTypes.TokenDataId,
+				ValueType = ValueTypes.TokenData,
+				Key = new TokenDataId
+				{
+					Creator = creator,
+					Collection = collectionName,
+					Name = tokenName
+				}
+			};
+
+			return await _client.GetTableItem<TokenPayload>(handle, request);
+		}
+
+		public Task<TokenFromAccount> GetToken(
+			string creator,
+			string collectionName,
+			string tokenName,
+			long propertyVersion
+			)
+		{
+			var tokenId = new TokenId
+			{
+				TokenDataId = new TokenDataId
+				{
+					Creator = creator,
+					Collection = collectionName,
+					Name = tokenName
+				},
+				PropertyVersion = propertyVersion.ToString()
+			};
+
+			return GetTokenForAccount(creator, tokenId);
+		}
+
+		public async Task<TokenFromAccount> GetTokenForAccount(string creator, TokenId tokenId)
+		{
+			TokenResource collectionData = null;
+			try
+			{
+				var collections = await _client.GetAccountResource(creator, ResourcesTypes.TokenStore);
+				collectionData = collections.Data.ToObject<TokenResource>();
+			}
+			catch (AptosException e)
+			{
+				if (e.ErrorCode == "resource_not_found")
+				{
+					return CreateEmptyToken(tokenId);
+				}
+			}
+
+			var handle = collectionData.Tokens.Handle;
+			
+			var request = new TableItemRequest
+			{
+				KeyType = KeyTypes.TokenId,
+				ValueType = ValueTypes.Token,
+				Key = tokenId
+			};
+			
+			try
+			{
+				return await _client.GetTableItem<TokenFromAccount>(handle, request);
+			}
+			catch (AptosException e)
+			{
+				Console.WriteLine(e.ErrorCode);
+				if (e.ErrorCode == "table_item_not_found")
+				{
+					return CreateEmptyToken(tokenId);
+				}
+
+				throw;
```

Comment:
do you handle rethrown exceptions up the call stack?

Reply:
No, but I think it should be rethrown, because only table_item_not_found exception can be interpreted in a certain way.